src/order_gui.cpp

```diff
@@ -555,6 +555,16 @@ struct OrdersWindow : public Window {
 		return (sel <= vehicle->GetNumOrders() && sel >= 0) ? sel : INVALID_VEH_ORDER_ID;
 	}
 
+	/**
+	 * Handle the click on the automatic separation button
+	 */
+	void OrderClick_AutomaticSeparation()
+	{
+		if (DoCommandP(this->vehicle->tile, this->vehicle->index, !this->vehicle->AutomaticSeparationIsEnabled(), CMD_ORDER_AUTOMATIC_SEPARATION | CMD_MSG(STR_ERROR_CAN_T_DELETE_THIS_ORDER))) {
```

Comment:
The error message seems wrong for this command?

Reply:
Ah, I forgot this. But I am also not sure what error message, if any, should go here. The command can only fail if you don't own the vehicle or try and run it on a non-primary vehicle, neither of which I believe to be possible through the GUI.

For now, I've removed the error message here, but please do let me know how this should be handled if that's not right.